src/data/nat/factorization/basic.lean

```diff
@@ -481,6 +481,16 @@ begin
   { simp [hqp, (factorization_le_iff_dvd hd0 hn0).2 hdn q] },
 end
 
+/-- If `n` is a nonzero natural number and `p ≠ 1`, then there are natural numbers `e`
+and `n'` such that `n'` is not divisible by `p` and `n = p^e * n'`. -/
+lemma exists_prime_pow_and_compl {n : ℕ} (hn : n ≠ 0) (p : ℕ) (hp : p ≠ 1) :
+  ∃ e n' : ℕ, ¬ p ∣ n' ∧ n = p ^ e * n' :=
+begin
+  obtain ⟨a', h₁, h₂⟩ := multiplicity.exists_eq_pow_mul_and_not_dvd
+                           (multiplicity.finite_nat_iff.mpr ⟨hp, nat.pos_of_ne_zero hn⟩),
+  refine ⟨_, a', h₂, h₁⟩,
+end
```

Comment:
`exact` also works. But why not
```suggestion
let ⟨a', h₁, h₂⟩ := multiplicity.exists_eq_pow_mul_and_not_dvd
                      (multiplicity.finite_nat_iff.mpr ⟨hp, nat.pos_of_ne_zero hn⟩) in
⟨_, a', h₂, h₁⟩
```

Reply:
OK, slightly shorter. The name should also change and not mention "prime". Is `exists_pow_and_compl` OK?

Reply:
I'd suggest nat.exists_eq_pow_mul_and_not_dvd as it's just a specialization of the multiplicity lemma.

Reply:
Sounds reasonable. I'll wait to see if CI turns up any problems and then change it.